### PR TITLE
AZ-578 Add fame_azure-vpn detectors

### PR DESCRIPTION
**Pull request by jmapro**
Create new detector for Azure VPN based on FAME metrics

### CODE REVIEW

File: modules/fame_azure-vpn/filters.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+locals {
```
Comment (BzSpi) [inline modules/fame_azure-vpn/filters.tf:1]: The filters should be a link to https://github.com/claranet/terraform-signalfx-detectors/blob/master/common/module/filters-integration-azure.tf

**Review by BzSpi — CHANGES_REQUESTED** [review]

**Review by Shr3ps — APPROVED** [review]

File: modules/fame_azure-vpn/filters.tf (hunk 1 above)
Comment (jmapro) [reply modules/fame_azure-vpn/filters.tf]: As discussed internally, I've reverted the link to filters-integration-azure. FAME act as an agent so it must use non prefixed tags.

File: modules/fame_azure-vpn/conf/01-total-flow-count.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module: Azure VPN
+name: Total Flow Count
+id: totalflowcount
+
+transformation: true
```
Comment (BzSpi) [inline modules/fame_azure-vpn/conf/01-total-flow-count.yaml:5]: A transformation is needed here. I think that "max" is a good one.

Comment (BzSpi) [inline modules/fame_azure-vpn/conf/01-total-flow-count.yaml:5]: A transformation is needed here. I think that "max" is a good one.

**Review by BzSpi — CHANGES_REQUESTED** [review]

**Review by BzSpi — APPROVED** [review]

File: modules/fame_azure-vpn/conf/01-total-flow-count.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+aggregation: true
+
+signals:
+  signal:
+    metric: fame.azure.virtual_network_gateway.total_flow_count
+
+rules:
+  critical:
+    threshold: 0
+    comparator: "=="
```
Comment (Shr3ps) [inline modules/fame_azure-vpn/conf/01-total-flow-count.yaml:15]: open discussion: isn't it better "< 1"?

**Review by Shr3ps — APPROVED** [review]

File: modules/fame_azure-vpn/conf/01-total-flow-count.yaml (hunk 3 above)
Comment (jmapro) [reply modules/fame_azure-vpn/conf/01-total-flow-count.yaml:15]: no, because 0.5 is a valid value.

**Pull request merged**